src/Microsoft.Extensions.Logging.AzureAppServices/Internal/FileLoggerProvider.cs

```diff
@@ -1,98 +1,80 @@
+                    }
+                }
             }
 
-            _fileSizeLimit = fileSizeLimit;
-            _retainedFileCountLimit = retainedFileCountLimit;
-            _backgroundQueueSize = backgroundQueueSize;
-            _outputTemplate = outputTemplate;
-            _flushPeriod = flushPeriod;
+            RollFiles();
         }
 
-        /// <inheritdoc />
-        public Logger ConfigureLogger(IWebAppLogConfigurationReader reader)
+        private string GetFullName((int Year, int Month, int Day) group)
         {
-            var webAppConfiguration = reader.Current;
-            if (string.IsNullOrEmpty(webAppConfiguration.FileLoggingFolder))
-            {
-                throw new ArgumentNullException(nameof(webAppConfiguration.FileLoggingFolder),
-                    "The file logger path cannot be null or empty.");
-            }
-
-            var logsFolder = webAppConfiguration.FileLoggingFolder;
-            if (!Directory.Exists(logsFolder))
-            {
-                Directory.CreateDirectory(logsFolder);
-            }
-            var logsFilePattern = Path.Combine(logsFolder, FileNamePattern);
+            return Path.Combine(_path, $"{_fileName}{group.Year:0000}{group.Month:00}{group.Day:00}.txt");
+        }
 
-            var messageFormatter = new MessageTemplateTextFormatter(_outputTemplate, null);
-            var rollingFileSink = new RollingFileSink(logsFilePattern, messageFormatter, _fileSizeLimit, _retainedFileCountLimit);
+        public (int Year, int Month, int Day) GetGrouping(LogMessage message)
+        {
+            return (message.Timestamp.Year, message.Timestamp.Month, message.Timestamp.Day);
+        }
 
-            ILogEventSink flushingSink;
-            if (_flushPeriod != null)
-            {
-                flushingSink = new PeriodicFlushToDiskSink(rollingFileSink, _flushPeriod.Value);
-            }
-            else
+        protected void RollFiles()
+        {
+            if (_maxRetainedFiles > 0)
             {
-                flushingSink = rollingFileSink;
-            }
-            var backgroundSink = new BackgroundSink(flushingSink, _backgroundQueueSize);
+                var files = new DirectoryInfo(_path)
+                    .GetFiles(_fileName + "*")
+                    .OrderByDescending(f => f.Name)
+                    .Skip(_maxRetainedFiles.Value);
 
-            var loggerConfiguration = new LoggerConfiguration();
-            loggerConfiguration.WriteTo.Sink(backgroundSink);
-            loggerConfiguration.MinimumLevel.ControlledBy(new WebConfigurationReaderLevelSwitch(reader,
-                configuration =>
+                foreach (var item in files)
                 {
-                    return configuration.FileLoggingEnabled ? configuration.FileLoggingLevel : LogLevel.None;
-                }));
-            return loggerConfiguration.CreateLogger();
+                    item.Delete();
```

Comment:
Handle errors gracefully. All sorts of mayhem can happen here (file in use, security, etc).